TotalCrossSDK/src/main/java/totalcross/ui/Switch.java

```diff
@@ -203,35 +182,16 @@ private void buildBall() throws Exception {
     }
   }
 
-  private void buildBar() throws Exception {
+  private void buildBar(boolean on) throws Exception {
     int h = uiMaterial ? height / 3 * 2 : height;
+    int barY = (height - h)/2;
     int fillB = !isEnabled() ? Color.interpolate(backColor, parent.backColor) : backColor;
-    Image barLR = (uiMaterial ? Resources.switchBrd : isIos ? Resources.switchBrdIos : Resources.switchBrdAnd)
-    		.smoothScaledFixedAspectRatio(uiMaterial || isIos ? h : h - 2, true); // left/right
-    int bw = barLR.getWidth() / 2;
-    Image barMid = uiMaterial ? null : Resources.switchBack.getSmoothScaledInstance(width - 2 * bw, h); // mid
-    barOn = new Image(width, h);
-    
-    Graphics gg = barOn.getGraphics();
-    if (!uiMaterial) {
-      gg.drawImage(barMid, bw, isIos ? 0 : -1);
-      
-      // draw left
-	  gg.setClip(0, 0, bw, h);
-	  gg.drawImage(barLR, 0, 0);
-	  
-	  // draw right
-	  gg.setClip(width - bw - 1, 0, bw, h);
-	  gg.drawImage(barLR, width - 2 * bw - 1, 0);
-	  gg.clearClip();
-    } else {
-      gg.backColor = Color.WHITE;
-      gg.fillRoundRect(2, 0, width - 2, h, h);
-      barOff = barOn.getCopy();
-      barOff.applyColor2(uiMaterial ? colorBarOff : fillB);
-    }
-    
-    barOn.applyColor2(uiMaterial ? colorBarOn : fillB);
+    Graphics g = getGraphics();
+    int gap = UnitsConverter.toPixels(DP + 2);
```

Comment:
this gap must not be settable by the user. It's just that the bar must end before the edge of the component. The reason why I do this is because the bar appear bellow the "holder" if I let it go until the edge.